src/lifx/light.js

```diff
@@ -529,4 +529,159 @@ Light.prototype.colorZones = function(startIndex, endIndex, hue, saturation, bri
+    ret.length = 8;
+  }
+  if (typeof ret.x !== 'number') {
+    ret.x = 0;
+  }
+  if (typeof ret.y !== 'number') {
+    ret.y = 0;
+  }
+  if (typeof ret.duration !== 'number') {
+    ret.duration = 8;
+  }
+  return ret;
+}
+
+/**
+ * Requests tile GetTileState64 707
+ *
+ * Get the state of 64 pixels in the tile in a rectangle that has
+ * a starting point and width.
+ * The tileIndex is used to control the starting tile in the chain
+ * and length is used to get the state of that many tiles beginning
+ * from the tileIndex. This will result in a separate response from
+ * each tile.
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Object} options - options passed to
+ * @param {Number} options.length	unsigned 8-bit integer (default 64)
+ * @param {Number} options.x	unsigned 8-bit integer (default 0)
+ * @param {Number} options.y	unsigned 8-bit integer (default 0)
+ * @param {Number} options.width	unsigned 8-bit integer (default 8)
+ * @param {Number} options.reserved	unsigned 8-bit integer
+ * @param {Function} callback a function to accept the data
+ */
+Light.prototype.getTileState64 = function(tileIndex, options, callback) {
+  validate.isUint8(tileIndex, 'getTileState64', 'tileIndex');
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  validate.callback(callback, 'light getTileState64 method');
+  const {length, x, y, width, reserved} = defaultOptionsTileState64(options);
+  const packetObj = packet.create('getTileState64', {
+    tileIndex,
+    length,
+    reserved: reserved || 0,
+    x,
+    y,
+    width
+  },
+  this.client.source
+  );
+  packetObj.target = this.id;
+  const sqnNumber = this.client.send(packetObj);
+  this.client.addMessageHandler('stateTileState64', function(err, msg) {
+    if (err) {
+      return callback(err, null);
+    }
+    return callback(null, msg);
+  }, sqnNumber);
+};
+
+/**
+ * This lets you set 64 pixels from a starting x and y for
+ * a rectangle with the specified width.
+ * For the LIFX Tile it really only makes sense to set x
+ * and y to zero, and width to 8.
+ * @param {Number} tileIndex	unsigned 8-bit integer
+ * @param {Number} colors[64]	64 HSBK values
+ * @param {Object} options - options passed to
+ * @param {Number} options.length	unsigned 8-bit integer (default 0)
+ * @param {Number} options.x	unsigned 8-bit integer (default 8)
+ * @param {Number} options.y	unsigned 8-bit integer (default 8)
+ * @param {Number} options.width	unsigned 8-bit integer (default 8)
+ * @param {Number} options.duration	unsigned 32-bit integer (default 0)
+ * @param {Number} options.reserved	unsigned 8-bit integer
+ * @param {Function} [callback] called when light did receive message
+ */
+Light.prototype.setTileState64 = function(tileIndex, colors, options, callback) {
+  validate.isUint8(tileIndex, 'setTileState64', 'tileIndex');
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
```

Comment:
What's the purpose of the above block? Also note that `options` is assigned to `{}` but `options` is not used after this.

Reply:
this was a typo